updated program year leadership label

dartajax opened the pull request:
```
On branch update_program_year_translation_label
Changes to be committed:
        modified:   app/components/program-year/details.hbs
        modified:   tests/acceptance/program-year/leadership-test.js
        modified:   translations/en-us.yaml
        modified:   translations/es.yaml
        modified:   translations/fr.yaml
```

* modified `details.hbs` to point to the newly added `leadership` translation - using this instead of `programYearLeadership`. 
* updated all three translation files by adding this new value
* updated test coverage to look for this new value 

There is certainly no hurry on this - feel free to discuss if even necessary however I propose that it makes the screen look cleaner since we don't pre-pend "Program Year" on the other values below. 

This would close https://github.com/ilios/ilios/issues/5062 if approved and merged.

Related to https://github.com/ilios/ilios/issues/5063 in the sense that it is dealing with the same caption on program year detail. 

Here is what the change looks like in English - Spanish and French also updated.

![image](https://github.com/ilios/frontend/assets/7493331/2292415d-bf3e-4002-89c8-0d2e61368a2e)

stopfstedt reviewed (changes requested):
looks great so far, but also please switch out the label on the leadership-expanded component.

![image](https://github.com/ilios/frontend/assets/1410427/8d3b565f-1629-4b81-842f-3a59fa2a8397)

`app/components/program-year/leadership-expanded.hbs` would be the right place.

stopfstedt commented:
@dartajax - while you're in there, please allow me to add a bit of scope creep. 

please apply this change to "Program Leadership" as well.

see `app/components/program/leadership-expanded.hbs` and `app/components/program/root.hbs`

dartajax commented:
@stopfstedt I did the Program level stuff - Title (short) and Leadership but I still need to check on the exact thing you mentioned in your comment so sorry about that

dartajax commented:
latest commit and PR -- 

 updated program year leadership label #7510 

```
On branch update_program_year_translation_label
Your branch is up to date with 'origin/update_program_year_translation_label'.

Changes to be committed:
        modified:   tests/acceptance/program-year/leadership-test.js
        modified:   tests/acceptance/program/overview-test.js
        modified:   tests/integration/components/program/leadership-expanded-test.js```

dartajax commented:
fixes https://github.com/ilios/ilios/issues/5069

stopfstedt reviewed (approved):
LGTM
